app/src/main/java/com/spascoding/taskycourse/feature_auth/presentation/util/NameInitialsExtractor.kt

```diff
@@ -0,0 +1,33 @@
+package com.spascoding.taskycourse.feature_auth.presentation.util
+
+class NameInitialsExtractor(val name: String) {
+
+    fun extract(): String {
+        return when(wordCount(name)) {
+            0 -> throw IllegalArgumentException("Name must contains minimum two letters")
+            1 -> extractFromOneWord()
+            else -> extractFromMoreThanTwoWords()
+        }
+    }
+
+    private fun extractFromOneWord(): String {
+        return "${name[0]}${name[1]}".uppercase()
+    }
+
+    private fun extractFromMoreThanTwoWords(): String {
+        val words = name.split("\\s+".toRegex())
+        val firstName = words[0]
+        val secondName = words[1]
+        return "${firstName[0]}${secondName[0]}".uppercase()
+    }
```

Comment:
Based on the requirements, a full name with three words should result in the initials of the first and last name, not the first and second